### DISCUSSION TITLE
ModalSubmitInteraction returning undefined SelectMenu values

### DISCUSSION THREAD

**Discussion by mattmular** [discordjs/discord.js#8005] [Q&A] [answered]
Wondering if anyone can help me or tell me what I'm doing wrong.

**Modal Object**
```
new Modal()
    .setCustomId('SubmitTicket')
    .setTitle('Support Ticket')
    .addComponents(new MessageActionRow()
        .addComponents(new MessageSelectMenu()
            .setCustomId('TicketType')
            .setPlaceholder('Appeal Type')
            .setMinValues(1)
            .setMaxValues(1)
            .addOptions([{label:'Appeal',value:'APPEAL'},{label:'Report',value:'REPORT'},{label:'Other',value:'OTHER'}])
        ));
```

**Interaction Handler**
```
client.on('interactionCreate', interaction => {
    if (!interaction.isModalSubmit()) return;

    if (interaction.customId.includes('SubmitTicket')) {
        console.log(interaction.fields.getField('TicketType'));
        interaction.reply('received');
    }
});
```

**Modal in application**
![Screenshot 2022-06-04 222326](https://user-images.githubusercontent.com/32026066/172036838-7274fa26-d6ec-4f16-9601-22724a5c50eb.png)

**Output**
```
C:\Program Files\nodejs\node.exe .\index.js
{value: undefined, type: 'SELECT_MENU', customId: 'TicketType'}
```

It works fine with TextInputs, I have not tested with other component types
`{value: 'This is text submitted through a modal interaction', type: 'TEXT_INPUT', customId: 'TicketComment'}`

**Comment by megatank58** [comment, marked as answer]
discord.js does not has support for select menus in modals yet, #7974

**Reply by monbrey** [reply]
To add - technically neither does Discord. This is an unannounced and undocumented feature in the API that isn't ready yet

**Comment by gpared** [comment]
Some release to support select menus in modal ?

**Reply by jaw0r3k** [reply]
ask discord

**Comment by HemendraKhatik** [comment]
@mattmular any update on this?

**Reply by megatank58** [reply]
Discord has not released it yet so there is nothing that can be done as far as i know.

**Reply by Sasino97** [reply]
What about now? :)
